src/Fixer/Whitespace/NoSpacesBetweenOffsetFixer.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class NoSpacesBetweenOffsetFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(array('[', CT_ARRAY_INDEX_CURLY_BRACE_OPEN));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        /* @var $token \PhpCsFixer\Tokenizer\Token */
+        foreach ($tokens as $index => $token) {
+            if (!$token->equalsAny(array(new Token('['), new Token(array(CT_ARRAY_INDEX_CURLY_BRACE_OPEN, '{'))))) {
```

Comment:
If I use `{` instead of `new Token(array(CT_ARRAY_INDEX_CURLY_BRACE_OPEN, '{'))`, the fixer fails:
```
--- Expected
+++ Actual
@@ @@
 <?php
-$foo{0}{1}{2} = 3;
+$foo {0} {1}   {2} = 3;
```
Maybe you have a better approach for this situation, really I don't know how this happens.